pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1094,6 +1103,23 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		}
 	}
 
+	// Add Ignition Command Line if present
+	ignitiondata, _ := vmi.Annotations["kubevirt.io/ignitiondata"]
+	if ignitiondata != "" && strings.Contains(ignitiondata, "ignition") {
+		virtconfig.Init()
+		if virtconfig.IgnitionEnabled() {
```

Comment:
Ah, this won't work because the virt-launcher can't retrieve the kubevirt-config config map. 

The easiest thing to do is to just add a check for the ignition annotation in the validation webhook.

```pkg/virt-api/webhooks/validating-webhook/validating-webhook.go```

There will need to be a check in both the admitVMICreate and ValidateVirtualMachineSpec functions to ensure the feature flag is enabled if that annotation is set. If the annotation is set but the feature flag isn't enabled, the validation webhook prevents the VM/VMI from ever getting posted to begin with.

so, with the validation webhook check, we don't need any more validation. The code in virt-launcher can just assume it should attempt ignition support if it sees the annotation.

Reply:
thanks for the pointers, had to create a dedicated function though since the annotation lives in the virtualmachineinstance metadata (and not spec)